SingularityUI/app/components/requests/RequestsPage.jsx

```diff
@@ -42,6 +60,18 @@ class RequestsPage extends React.Component {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (this.props.params !== nextProps.params) {
+      this.setState({
+        filter: {
+          state: nextProps.params.state || 'all',
+          subFilter: !nextProps.params.subFilter || nextProps.params.subFilter === 'all' ? RequestFilters.REQUEST_TYPES : nextProps.params.subFilter.split(','),
+          searchFilter: nextProps.params.searchFilter || ''
+        }
+      });
+    }
+  }
```

Comment:
it seems like this page is trying to manage some complex state that's coming in from props passed by the router -- are there other ways we can handle this?